addons/sale_project/models/sale_order.py

```diff
@@ -119,13 +119,13 @@ def action_view_task(self):
             action['views'] = [(form_view_id, 'form')]
             action['res_id'] = self.tasks_ids.id
         # set default project
-        default_line = next(sol for sol in self.order_line if sol.product_id.detailed_type == 'service')
-        default_project_id = default_line.project_id.id or self.project_id.id or self.project_ids[:1].id
+        default_line = next((sol for sol in self.order_line if sol.product_id.detailed_type == 'service'), None)
```

Comment:
```suggestion
        default_line = next((sol for sol in self.order_line if sol.product_id.detailed_type == 'service'), self.env['sale.order.line']])
```
Isn't an empty set a better "default"? Moreover L123 and L128 wouldn't be needed anymore.

Reply:
Hello @auon-odoo ,
Thanks for you suggestion,
i have perform the changes